EESSI Governance (#456)
EESSI/docs: docs/governance/charter.md

<!-- 
A project charter discusses _what it is and why it exists_, a governance discusses _how it operates_.

Examples

https://docs.clearlydefined.io/docs/community/charter
https://openssf.org/about/charter/
https://github.com/cncf/foundation/blob/main/charter.md
https://github.com/mochajs/mocha/blob/main/PROJECT_CHARTER.md
https://github.com/nodejs/TSC/blob/main/TSC-Charter.md
Combined charter - governance https://github.com/camaraproject/Governance/blob/main/ProjectCharter.md
blog about charters https://opensource.org/blog/what-is-open-governance-drafting-a-charter-for-an-open-source-project
-->

# Project Charter

## 1. Mission
<!-- Describe the project's purpose and the problem it addresses. Include a short mission statement. -->

> The EESSI project aims to build a common software stack that is:

Reviewer: Spell out EESSI in full the first time you use it + make it link to EESSI homepage?

Maybe also include logo on top of the page

> - Cross-platform (laptop, Cloud VM, HPC Cluster)

Reviewer: Needs empty line above to fix formatting

- Ready-to-use (served over the internet, just mount-and-go)
- Optimized for a wide range of hardware architectures (CPU, GPU, interconnects)

> - Easily extendable

Reviewer: Should clarify what this means? Maybe also "customizable" (cfr. host injections)

Extra bullet for focus on Linux systems/VMs? (no native macOS/Windows)

## 2. Scope
<!-- Define what is within the scope of the project and what is explicitly out of scope. -->

> EESSI will focus on creating a repository of software installations (software.eessi.io). This requires:

Reviewer: `repository` is common technical terminology that usually means "package repository" (like RPMs), but actually means something else for us (CernVM-FS repository), so let's avoid that term (or be explicit), something like:
```suggestion
EESSI focuses on creating a shared stack of software installations (`software.eessi.io`) that is distributed through a [CernVM-FS](https://cernvm.cern.ch/fs/) repository.

This requires:
```

> - code to build and deploy new software into the repository

Reviewer: Needs empty line above to fix formatting + capitalize bullet points

> - code to make EESSI work on end-user systems

Reviewer: code -> scripts

"work" -> user-friendly interface?

- infrastructure to build new software for software.eessi.io

> - infrastructure to host the repository for software.eessi.io

Reviewer: CernVM-FS repository

All of these (both code and infrastructure itself) are considered 'in scope' for the project.

> There are additional repositories under the eessi.io namespace, such as dev.eessi.io and riscv.eessi.io. All code related to those repositories is also considered part of the EESSI project. However, the repositories themselves (i.e. their content) and the infrastructure hosting them is not.

Reviewer: ```suggestion
There are additional CernVM-FS repositories under the `eessi.io` namespace, such as `dev.eessi.io` and `riscv.eessi.io`.
All code related to those repositories is also considered part of the EESSI project.
However, the repositories themselves (i.e. their content) and the infrastructure hosting them is not.
```

Author: Agreed, but we should say this in the governance (the charter just discusses what is part of  the EESSI open source project - and we consider this part of EESSI, so I now included it).

## 3. Membership
<!-- Who can join or participate? Are there any requirements (e.g., code contributions, voting eligibility)? -->

> There is currently no membership. Any individual or institution may participate by using the EESSI software stack, contributing to the EESSI software stack, making the EESSI software stack available on systems managed by them, etc.

Reviewer: ```suggestion
There is currently no membership. Any individual or institution may participate by using EESSI, contributing to EESSI, making EESSI available on systems managed by them, etc.

## 4. Review and Amendment

> Changes to the charter require approval by the Steering Committee

Reviewer: Link to relevant section on the policy we have for this?

- Source code to automate the process of building and deploying additional software installations in `software.eessi.io`
- Source code to provide a user-friendy interface on end-user systems
- infrastructure to build new software for software.eessi.io
- infrastructure to host the CernVM-FS repository for software.eessi.io

All of these (both code and infrastructure itself) are considered 'in scope' for the project.

There are additional CernVM-FS repositories under the eessi.io namespace, such as dev.eessi.io and riscv.eessi.io. All code and infrastructure related to those repositories (and any other CernVM-FS repositories under the eessi.io namespace) are also considered part of the EESSI project.

## 3. Membership
<!-- Who can join or participate? Are there any requirements (e.g., code contributions, voting eligibility)? -->
There is currently no membership. Any individual or institution may participate by using EESSI, contributing to EESSI, making EESSI available on systems managed by them, etc.

## 4. Review and Amendment

> Changes to the charter require approval by the Steering Committee. See the [relevant section of the Governance](../governance.md#voting-by-the-steering-committee).

Reviewer: ```suggestion
Changes to the charter require approval by the Steering Committee. See the [relevant section of the Governance](governance.md#voting-by-the-steering-committee).

<!-- 

> A project charter discusses _what it is and why it exists_, a governance discusses _how it operates_.

Author: aaah oh wait, my bad. You meant the word 'a' should be omitted. Clearly, me not being a native speaker is the issue here :) I guess I meant to refer to _the document that describes the governance_ (which could be 'a'), but that's not what it says here :)
